sphinxgallery/gen_rst.py

```diff
@@ -373,7 +387,7 @@ def execute_script(code_block, example_globals, image_path, fig_count,
     cwd = os.getcwd()
     # Redirect output to stdout and
     orig_stdout = sys.stdout
-
+    figure_list = []
```

Comment:
> I guess that means that examples without plots are currently broken ...

Examples without plots do get executed and render very well, there are 2 of them, I don't see them broken. The function save_figures does take care of giving an empty list if there are no plots.

What I see, is that this avoids a NameError if the code block did not execute correctly, leaving you out of the try-catch without the variable figure_list that you later need in the return statement. Maybe move this inside the Exception block?